src/main/java/org/broadinstitute/consent/http/models/Dataset.java

```diff
@@ -243,6 +243,21 @@ public static String parseAliasToIdentifier(Integer alias) {
         return PREFIX + StringUtils.leftPad(alias.toString(), 6, "0");
     }
 
+    public static Integer parseIdentifierToAlias(String identifier) throws IllegalArgumentException {
+        try {
+            String givenPrefix = identifier.substring(0, PREFIX.length());
+            if (!givenPrefix.equals(PREFIX)) {
+                throw new IllegalArgumentException("Invalid prefix.");
+            }
+
+            String aliasAsString = identifier.substring(PREFIX.length()); // cut off DUOS-
+            return Integer.parseInt(aliasAsString); // parse remaining as integer
```

Comment:
One thing I see in testing, these two identifiers are in effect equal: `DUOS-5` and `DUOS-000005`. It doesn't make any practical difference, but perhaps we could do a check in the resource class to ensure that the dataset returned actually has the same `datasetIdentifier` that the user passed in.